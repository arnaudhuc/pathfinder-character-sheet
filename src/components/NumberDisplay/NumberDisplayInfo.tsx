/** @jsx jsx */
import { jsx } from '@emotion/core';

interface NumberDisplayInfoLeft {
	message: string | number;
	isLeftDark: boolean;
	isLeft: boolean;
}

const getBackgroundColor = (isLeft: boolean, isLeftDark: boolean): string => {
	if (isLeftDark) return '#000';
	if (!isLeft) return '#fff';
	return '#bdbdbd';
};

const border = (isLeft: boolean) => {
	if (!isLeft) {
		return {
			border: '2px solid black',
		};
	}
};

export const NumberDisplayInfo: React.FC<NumberDisplayInfoLeft> = (props) => {
	const { message, isLeftDark, isLeft } = props;
	return (
		<div
			css={{
				backgroundColor: getBackgroundColor(isLeft, isLeftDark),
				width: isLeft ? '50px' : '46px',
				height: isLeft ? '50px' : '46px',
				transform: 'rotate(45deg)',
				position: 'absolute',
				left: isLeft ? '-28px' : '46px',
				top: '48px',
				...border(isLeft),
			}}
		>
			<div
				css={{
					transform: 'rotate(-45deg)',
					color: isLeft ? '#fff' : '#000',
					width: isLeft ? '50px' : '46px',
					height: isLeft ? '50px' : '46px',
					display: 'flex',
					justifyContent: 'center',
					alignItems: 'center',
				}}
			>
				<span>{message}</span>
			</div>
		</div>
	);
};
