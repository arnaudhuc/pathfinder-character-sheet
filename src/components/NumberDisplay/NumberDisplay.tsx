/** @jsx jsx */
import { jsx } from "@emotion/core";
import { NumberDisplayInfo } from "./NumberDisplayInfo";

interface NumberDisplay {
  numberToDisplay: number;
  messageLeft?: string | number;
  messageRight?: string | number;
  isLeftDark?: boolean;
}

export const NumberDisplay: React.FC<NumberDisplay> = (props) => {
  const {
    messageLeft,
    messageRight,
    isLeftDark = false,
    numberToDisplay,
  } = props;

  return (
    <div
      css={{
        width: "70px",
        height: "70px",
        transform: "rotate(45deg)",
        border: "7px solid",
        borderColor: isLeftDark ? "#bdbdbd" : "black",
        position: "relative",
      }}
    >
      <div
        css={{
          width: "70px",
          height: "70px",
          transform: "rotate(-45deg)",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <span>{numberToDisplay}</span>
        {messageLeft && (
          <NumberDisplayInfo
            isLeft={true}
            message={messageLeft}
            isLeftDark={isLeftDark}
          />
        )}
        {messageRight && (
          <NumberDisplayInfo
            isLeft={false}
            message={messageRight}
            isLeftDark={false}
          />
        )}
      </div>
    </div>
  );
};
