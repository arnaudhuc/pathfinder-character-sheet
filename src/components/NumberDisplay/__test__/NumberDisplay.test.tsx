/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';
import { NumberDisplay } from '../NumberDisplay';

describe('NumberDisplay', () => {
	test('should render with default props', () => {
		const container = renderer
			.create(<NumberDisplay messageLeft="CA" numberToDisplay={12} />)
			.toJSON();

		expect(container).toMatchSnapshot();
	});
	test('should render with right value', () => {
		const container = renderer
			.create(
				<NumberDisplay
					messageLeft="CA"
					numberToDisplay={12}
					isLeftDark={true}
					messageRight={'-5'}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});
});
