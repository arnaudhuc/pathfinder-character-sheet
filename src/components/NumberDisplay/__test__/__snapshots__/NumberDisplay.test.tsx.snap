// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberDisplay should render with default props 1`] = `
<div
  className="css-1xdfrjn-NumberDisplay"
>
  <div
    className="css-1oz0q46-NumberDisplay"
  >
    <span
      className="css-1gp7egd-NumberDisplay"
    >
      12
    </span>
    <div
      className="css-4mbg8w-NumberDisplayInfo"
    >
      <div
        className="css-bl4yr9-NumberDisplayInfo"
      >
        <span>
          CA
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberDisplay should render with right value 1`] = `
<div
  className="css-7zvqsh-NumberDisplay"
>
  <div
    className="css-1oz0q46-NumberDisplay"
  >
    <span
      className="css-1gp7egd-NumberDisplay"
    >
      12
    </span>
    <div
      className="css-17cu81j-NumberDisplayInfo"
    >
      <div
        className="css-o0gr1t-NumberDisplayInfo"
      >
        <span>
          CA
        </span>
      </div>
    </div>
    <div
      className="css-1n5dk3r-NumberDisplayInfo"
    >
      <div
        className="css-1yxj2i3-NumberDisplayInfo"
      >
        <span>
          -5
        </span>
      </div>
    </div>
  </div>
</div>
`;
